src/storage/txn/scheduler.rs

```diff
@@ -706,18 +776,43 @@ impl Scheduler {
         info!("receive shutdown command");
         event_loop.shutdown();
     }
+
+    fn register_roll_stats_tick(&self, event_loop: &mut EventLoop<Self>) {
+        if let Err(e) = register_timer(event_loop, Tick::RollStats, ROLL_INTERVAL_SECS * 1000) {
+            error!("register roll statistics tick err: {:?}", e);
+        };
+    }
+
+    fn on_roll_stats_tick(&mut self, event_loop: &mut EventLoop<Self>) {
+        self.write_stats.roll();
```

Comment:
seem that you can combine roll and incr_foreach together to avoid traversing all regions twice.